networkx/algorithms/swap.py

```diff
@@ -350,7 +350,7 @@ def connected_double_edge_swap(G, nswap=1, _window_threshold=3, seed=None):
             while wcount < window and n < nswap:
                 # Pick two random edges without creating the edge list. Choose
                 # source nodes from the discrete degree distribution.
-                (ui, xi) = nx.utils.discrete_sequence(2, cdistribution=cdf)
+                (ui, xi) = discrete_sequence(2, cdistribution=cdf, seed=seed)
```

Comment:
Won't this result in you always selecting the same pair of nodes, so you'd only be looking at swapped edges that originate from these two nodes? Maybe that's the intent within the "window"?

Reply:
Of course, I forgot about the decorator (a bit inexplicably given NXEP 4 :)